### PR TITLE
etcd: remove deprecated func and field

### PR DESCRIPTION
**Pull request by demoManito**

### CODE REVIEW

File: tests/e2e/ctl_v3_auth_test.go [hunk 1]
```diff
@@ -895,7 +895,7 @@ func leaseTestGrantLeasesList(cx ctlCtx) error {
 	if err != nil {
 		return fmt.Errorf("lease list failed (%v)", err)
 	}
-	_, err = proc.Expect(id)
+	_, err = proc.ExpectWithContext(context.Background(), id)
```
Comment (serathius) [inline tests/e2e/ctl_v3_auth_test.go:898]: Please use `context.TODO` instead. Context.Background() should be only used in top level of test.

File: tests/e2e/ctl_v3_auth_test.go [hunk 2]
```diff
@@ -1322,7 +1322,7 @@ func ctlV3User(cx ctlCtx, args []string, expStr string, stdIn []string) error {
 		}
 	}
 
-	_, err = proc.Expect(expStr)
+	_, err = proc.ExpectWithContext(context.Background(), expStr)
```
Comment (serathius) [inline tests/e2e/ctl_v3_auth_test.go:1325]: ditto.

Comment (demoManito) [reply tests/e2e/ctl_v3_auth_test.go]: modified @serathius

File: tests/functional/tester/stresser_lease.go [hunk 3]
```diff
@@ -135,7 +135,7 @@ func (ls *leaseStresser) Stress() error {
 	ls.ctx = ctx
 	ls.cancel = cancel
 
-	cli, err := ls.m.CreateEtcdClient(grpc.WithBackoffMaxDelay(1 * time.Second))
+	cli, err := ls.m.CreateEtcdClient(grpc.WithConnectParams(grpc.ConnectParams{MinConnectTimeout: 1 * time.Second}))
```
Comment (serathius) [inline tests/functional/tester/stresser_lease.go:138]:
```suggestion
	cli, err := ls.m.CreateEtcdClient(grpc.WithConnectParams(grpc.ConnectParams{Backoff: backoff.Config{MaxDelay: 1 * time.Second}}))
```

File: tests/functional/tester/stresser_key.go [hunk 4]
```diff
@@ -76,7 +76,7 @@ type keyStresser struct {
 
 func (s *keyStresser) Stress() error {
 	var err error
-	s.cli, err = s.m.CreateEtcdClient(grpc.WithBackoffMaxDelay(1 * time.Second))
+	s.cli, err = s.m.CreateEtcdClient(grpc.WithConnectParams(grpc.ConnectParams{MinConnectTimeout: 1 * time.Second}))
```
Comment (serathius) [inline tests/functional/tester/stresser_key.go:79]: ditto

File: tests/functional/tester/checker_short_ttl_lease_expire.go [hunk 5]
```diff
@@ -58,7 +58,7 @@ func (lc *shortTTLLeaseExpireChecker) Check() error {
 		return nil
 	}
 
-	cli, err := lc.m.CreateEtcdClient(grpc.WithBackoffMaxDelay(time.Second))
+	cli, err := lc.m.CreateEtcdClient(grpc.WithConnectParams(grpc.ConnectParams{MinConnectTimeout: time.Second}))
```
Comment (serathius) [inline tests/functional/tester/checker_short_ttl_lease_expire.go:61]: ditto

File: tests/functional/tester/checker_lease_expire.go [hunk 6]
```diff
@@ -63,7 +63,7 @@ func (lc *leaseExpireChecker) Check() error {
 		return nil
 	}
 
-	cli, err := lc.m.CreateEtcdClient(grpc.WithBackoffMaxDelay(time.Second))
+	cli, err := lc.m.CreateEtcdClient(grpc.WithConnectParams(grpc.ConnectParams{MinConnectTimeout: time.Second}))
```
Comment (serathius) [inline tests/functional/tester/checker_lease_expire.go:66]: ditto

**Comment by demoManito** [comment]
`Tests / test (linux-amd64-integration-4-cpu) (pull_request)`

This test has nothing to do with my modification, but it always fails. Can retry it manually? @serathius

**Comment by ahrtr** [comment]
Please rebase this PR.

**Comment by jmhbnz** [comment]
Hey @demoManito I'm doing some pr triage to hopefully get some older pr's merged - can you please rebase this pr to resolve conflicts?  Thanks! 🙏🏻

**Pull request closed without merging**